### DISCUSSION TITLE
Add a way to auto-prune revisions

### DISCUSSION THREAD

**Discussion by rijkvanzanten** [directus/directus#3117] [Feature Requests]
Revisions can grow pretty big pretty fast. We should have a way to auto-prune revisions, so it only saves the last _n_ revisions per item or something along those lines

@benhaynes

**Comment by benhaynes** [comment]
Yes, or _n_ days of revisions. When we implement this feature, it might be nice to have a way to "lock" certain key revisions so they don't get deleted or pruned.

**Comment by benhaynes** [comment]
Also, just a divergent idea, but if we have import/export maybe we can even allow for legacy revision data to be backed up elsewhere without losing the data but not clogging up the database.

**Comment by rijkvanzanten** [comment]
This feature request has been accepted and is queued to be implemented! You can follow along with the progress here: https://github.com/directus/directus/pull/18105

**Comment by JamesW1** [comment]
Note this can be configured using `REVISIONS_RETENTION` environment variable:
https://directus.io/docs/configuration/logging#log-retention
